runelite-client/src/main/java/net/runelite/client/util/MenuEntryItem.java

```diff
@@ -0,0 +1,131 @@
+	 * {@link WidgetInfo} entry exists for the widget.
+	 */
+	private WidgetInfo widget;
+	private int itemId;
+	private int itemQuantity;
+
+	/**
+	 * Gets an {@link MenuEntryItem} from an examined widget.
+	 *
+	 * @param widget      The examined widget
+	 * @param type        The examine type of the examination
+	 * @param actionParam The action param of the examination (see {@link MenuOptionClicked#getActionParam()} or
+	 *                    {@link MenuEntryAdded#getActionParam0()})
+	 * @return An {@link MenuEntryItem} of appropriate {@code id} and {@code quantity}, or -1 for both fields if the item
+	 *         id could not be found. Furthermore, if the passed {@code widget} is null, null will be returned.
+	 */
+	@Nullable
+	public static MenuEntryItem findExamineItem(final Widget widget, final ExamineType type, final int actionParam)
+	{
+		if (widget == null)
+		{
+			return null;
+		}
+
+		final int widgetId = widget.getId();
+		final int widgetGroup = WidgetInfo.TO_GROUP(widgetId);
+		WidgetInfo matchedWidgetInfo = null;
+		int foundItemId = -1;
+		int foundItemQuantity = -1;
+
+		if (type == ExamineType.ITEM)
+		{
+			WidgetItem widgetItem = widget.getWidgetItem(actionParam);
+			if (widgetItem != null)
+			{
+				foundItemQuantity = widgetItem.getQuantity();
+				foundItemId = widgetItem.getId();
+			}
+		}
+		else if (type == ExamineType.ITEM_BANK_EQ)
+		{
+			if (WidgetInfo.EQUIPMENT.getGroupId() == widgetGroup)
+			{
+				Widget widgetItem = widget.getChild(1);
+				if (widgetItem != null)
+				{
+					foundItemQuantity = widgetItem.getItemQuantity();
+					foundItemId = widgetItem.getItemId();
+				}
+			}
+			else if (WidgetInfo.SMITHING_INVENTORY_ITEMS_CONTAINER.getGroupId() == widgetGroup)
+			{
+				Widget widgetItem = widget.getChild(2);
+				if (widgetItem != null)
+				{
+					foundItemQuantity = widgetItem.getItemQuantity();
+					foundItemId = widgetItem.getItemId();
+				}
+			}
+			else if (WidgetInfo.BANK_INVENTORY_ITEMS_CONTAINER.getGroupId() == widgetGroup
+				|| WidgetInfo.RUNE_POUCH_ITEM_CONTAINER.getGroupId() == widgetGroup)
+			{
+				Widget widgetItem = widget.getChild(actionParam);
+				if (widgetItem != null)
+				{
+					foundItemQuantity = widgetItem.getItemQuantity();
+					foundItemId = widgetItem.getItemId();
+				}
+			}
+			else if (WidgetInfo.BANK_ITEM_CONTAINER.getGroupId() == widgetGroup
+				|| WidgetInfo.SHOP_ITEMS_CONTAINER.getGroupId() == widgetGroup
+				|| WidgetInfo.CLUE_SCROLL_REWARD_ITEM_CONTAINER.getGroupId() == widgetGroup
+				|| WidgetInfo.LOOTING_BAG_CONTAINER.getGroupId() == widgetGroup)
+			{
+				Widget[] children = widget.getDynamicChildren();
+				if (actionParam < children.length)
+				{
+					Widget widgetItem = children[actionParam];
+					foundItemQuantity = widgetItem.getItemQuantity();
+					foundItemId = widgetItem.getItemId();
+				}
+			}
+		}
+
+		if (foundItemQuantity != -1)
```

Comment:
checking for foundItemId would probably make more sense, also, I would just do early return null if foundItemId == -1.